src/test-run/index.js

```diff
@@ -530,6 +530,9 @@ export default class TestRun extends AsyncEventEmitter {
             }
 
             this._fulfillCurrentDriverTask(driverStatus);
+
+            if (driverStatus.isPendingWindowSwitching)
+                return new serviceCommands.DoNothingCommand();
```

Comment:
Why do we need to introduce a new command? We can respond `null` to the client driver if we don't have an actual command to execute. See https://github.com/DevExpress/testcafe/blob/master/src/test-run/index.js#L871

Reply:
The new command processed asynchronously (in the 'then' handler of the 'readyPromise').
If I will use the `return null` then it causes the [synchronous processing]( https://github.com/DevExpress/testcafe/pull/4598/files#diff-3690e674231a1353e109e26318442155L784).
It breaks the valid order of command extraction and execution.

Reply:
I've rewritten this PR without `doNothing` command.